TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -413,13 +422,30 @@ public static void ScanTokensLineContinuationGroup(IList<TokensLine> continuatio
 
                             // If an alphanumeric literal that is to be continued on the next line has as its last character a quotation mark in column 72, 
                             // the continuation line must start with two consecutive quotation marks.
+                            // In ColumnsLayout.CobolReferenceFormat we just check that the quotation mark is the last character of the line.
                             if (lastTokenOfConcatenatedLineSoFar.HasClosingDelimiter)
                             {
-                                if ((startOfContinuationIndex + 1) > lastIndex || line[startOfContinuationIndex + 1] != lastTokenOfConcatenatedLineSoFar.ExpectedClosingDelimiter)
+                                if (format == ColumnsLayout.CobolReferenceFormat 
+                                    ? (((startOfContinuationIndex + 1) <= lastIndex && line[startOfContinuationIndex + 1] == lastTokenOfConcatenatedLineSoFar.ExpectedClosingDelimiter) &&
+                                    (lastTokenOfConcatenatedLineSoFar.EndColumn + 8) != 73)
+                                    : ((startOfContinuationIndex + 1) > lastIndex || line[startOfContinuationIndex + 1] != lastTokenOfConcatenatedLineSoFar.ExpectedClosingDelimiter))
                                 {
-                                    continuationLine.AddDiagnostic(MessageCode.InvalidFirstTwoCharsForContinuationLine, startOfContinuationIndex, startOfContinuationIndex + 1, lastTokenOfConcatenatedLineSoFar.ExpectedClosingDelimiter);
+                                    continuationLine.AddDiagnostic(MessageCode.InvalidFirstTwoCharsForContinuationLine, startOfContinuationIndex, startOfContinuationIndex + 1, lastTokenOfConcatenatedLineSoFar.ExpectedClosingDelimiter,
+                                        format == ColumnsLayout.CobolReferenceFormat
+                                        ? "in column 72"
+                                        : "at the last column");
                                     // Use the first quotation mark to avoid a complete mess while scanning the rest of the line
                                     offsetForLiteralContinuation = 0;
+                                } else
+                                {
+                                    bool isQuoteInsertedInString = (((startOfContinuationIndex + 1) <= lastIndex && line[startOfContinuationIndex + 1] == lastTokenOfConcatenatedLineSoFar.ExpectedClosingDelimiter) &&
+                                        (format == ColumnsLayout.CobolReferenceFormat ? ((lastTokenOfConcatenatedLineSoFar.EndColumn + 8) == 73) : true));
```

Comment:
see my previous remarks about the format and the hardcoded values.

Reply:
changed in fe9d83c